### PR TITLE
Add Curl to Flash Player URL Processor

### PR DESCRIPTION
**Pull request by primalcurve**
Modified AdobeFlashURLProvider.py to utilize curl, using the core processor URLTextSearcher.py as a model. It now passes both the request_headers and curl_opts keys to the processor. This will mean the recipe has a greater chance for success in certain scenarios.

### CODE REVIEW

File: AdobeFlashPlayer/AdobeFlashURLProvider.py [hunk 1]
```diff
@@ -34,15 +35,25 @@ class AdobeFlashURLProvider(Processor):
     """Provides URL to the latest Adobe Flash Player release."""
     description = __doc__
     input_variables = {
-        "url": {
-            "required": False,
+        'url': {
```
Comment (gregneagle) [inline AdobeFlashPlayer/AdobeFlashURLProvider.py]: I don't feel strongly about single quotes vs double quotes, but consistency is nice. The existing code used double quotes here, so maybe your changes should as well.

File: AdobeFlashPlayer/AdobeFlashURLProvider.py [hunk 2]
```diff
@@ -34,15 +35,25 @@ class AdobeFlashURLProvider(Processor):
...
+            'required': False,
+        },
+        "CURL_PATH": {
             "required": False,
-            "description": ("Specific version to download. If not defined, "
-                            "defaults to latest version.")
+            "default": "/usr/bin/curl",
+            "description": "Path to curl binary. Defaults to /usr/bin/curl.",
         },
     }
```
Comment (gregneagle) [inline AdobeFlashPlayer/AdobeFlashURLProvider.py]: Why did you remove the "version" input variable? This will break people who use it.

File: AdobeFlashPlayer/AdobeFlashURLProvider.py (hunk 1 above)
Comment (primalcurve) [reply AdobeFlashPlayer/AdobeFlashURLProvider.py]: Okay. I can change that. You're right that consistency is good. I think my excitement at getting this working overrode my better judgement when reviewing my changes. I'll change them to double quotes.

File: AdobeFlashPlayer/AdobeFlashURLProvider.py (hunk 2 above)
Comment (primalcurve) [reply AdobeFlashPlayer/AdobeFlashURLProvider.py]: This is an oversight. I will fix it.

**Comment by primalcurve** [comment]
Changes submitted. I'm surprised I missed the version input variable. I was careful to keep its functionality in place otherwise.

**Comment by gregneagle** [comment]
@hjuutilainen or @timsutton Mind taking a second look here before we merge?

**Comment by primalcurve** [comment]
Oh whoops. I didn't realize it would continue updating. I have reverted the commits.
